lib/jsdom/living/validity/ValidityState-impl.js

```diff
@@ -0,0 +1,66 @@
+"use strict";
+
+exports.implementation = class ValidityStateImpl {
+  get badInput() {
+    return this.hasErrors("badInput");
+  }
+
+  get customError() {
+    return this.hasErrors("customError");
+  }
+
+  get patternMismatch() {
+    return this.hasErrors("patternMismatch");
+  }
+
+  get rangeOverflow() {
+    return this.hasErrors("rangeOverflow");
+  }
+
+  get rangeUnderflow() {
+    return this.hasErrors("rangeUnderflow");
+  }
+
+  get stepMismatch() {
+    return this.hasErrors("stepMismatch");
+  }
+
+  get tooLong() {
+    return this.hasErrors("tooLong");
+  }
+
+  get tooShort() {
+    return this.hasErrors("tooShort");
+  }
+
+  get typeMismatch() {
+    return this.hasErrors("typeMismatch");
+  }
+
+  get valueMissing() {
+    return this.hasErrors("valueMissing");
+  }
+
+  hasErrors(method) {
+    const validationMethod = this._state[method];
+      if (validationMethod) {
+          if (this._state.willValidate) {
+            return this._state.willValidate() && validationMethod();
+          } else {
+            return validationMethod();  
+        }
+    }
+
+    return false;  
+  }
+
+  constructor(state) {
+    this._state = state;
+  }
```

Comment:
1. Can you move this to the top of the class?
2. All IDL constructors follow the format `constructor(args, privateData)`. See https://github.com/jsdom/webidl2js#writing-implementation-class-files
3. Would it be better to hold a reference to the element, while making sure all element classes implement a specific interface (e.g. `_sufferingFrom`)? This way, we don't have to create a state object with its own callback functions for every element.